File: features/workflow/src/main/java/com/nivasafinance/features/workflow/dto/TaskResponse.java
package com.nivasafinance.features.workflow.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TaskResponse {
  
    private UUID taskIdentifier;

    private String taskConfigKey;

    private String taskName;

    private String taskDescription;

    private String assignedTo;

    private LocalDateTime dueAt;

    private String outcome;

    private OutcomeDetailsResponse outcomeDetails;

    private TaskDetailsResponse taskDetails;

    private LocalDateTime createdAt;

    private String createdBy;

    private LocalDateTime updatedAt;

    private String updatedBy;
}

File: features/tasks/src/main/java/com/nivasafinance/features/tasks/dto/TaskResponse.kt
package com.nivasafinance.features.tasks.dto

import java.time.LocalDateTime
import java.util.*

data class TaskResponse(
    val id: UUID,
    val taskDefinitionKey: String,
    val name: String,
    val taskType: String,
    val description: String?,
    val assignedTo: String?,
    val status: String,
    val outcome: String,
    val dueAt: LocalDateTime?,
    val completedAt: LocalDateTime?,
    val rescheduledAt: LocalDateTime?,
    val createdAt: LocalDateTime,
    val createdBy: String?,
    val updatedAt: LocalDateTime,
    val updatedBy: String?,
)
